packages/contact-center/task/tests/helper.ts

```diff
@@ -0,0 +1,215 @@
+    });
+
+    waitFor(() => {
+      expect(onDeclined).toHaveBeenCalled();
+      expect(taskMock.decline).toHaveBeenCalledWith('interaction1');
+    });
+  });
+
+  it('should correctly handle task media', async () => {
+    const taskMediaMock = {};
+    const {result} = renderHook(() =>
+      useIncomingTask({cc: ccMock, onAccepted, onDeclined, selectedLoginOption: 'BROWSER'})
+    );
+
+    // Simulate an incoming task
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock); // Trigger TASK_INCOMING event
+    });
+
+    // Simulate task media event
+    act(() => {
+      taskMock.on.mock.calls.find((call) => call[0] === 'task:media')[1](taskMediaMock);
+    });
+
+    waitFor(() => {
+      expect(result.current.missed).toBe(false);
+      expect(result.current.ended).toBe(false);
+    });
+  });
+});
+
+describe('useTaskList Hook', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should register task:incoming event and add task to the list', async () => {
+    const {result} = renderHook(() => useTaskList({cc: ccMock}));
+
+    // Simulate the incoming task event
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock); // Trigger TASK_INCOMING event
+    });
+
+    waitFor(() => {
+      expect(result.current.taskList).toContain(taskMock);
+      expect(ccMock.on).toHaveBeenCalledWith('task:incoming', expect.any(Function));
+    });
+  });
+
+  it('should handle multiple incoming tasks', async () => {
+    const {result} = renderHook(() => useTaskList({cc: ccMock}));
+
+    // Simulate multiple incoming task events
+    const task1 = {...taskMock, id: 'task1'};
+    const task2 = {...taskMock, id: 'task2'};
+
+    act(() => {
+      ccMock.on.mock.calls[0][1](task1); // Trigger TASK_INCOMING for task1
+      ccMock.on.mock.calls[0][1](task2); // Trigger TASK_INCOMING for task2
+    });
+
+    waitFor(() => {
+      expect(result.current.taskList).toEqual([task1, task2]);
+      expect(ccMock.on).toHaveBeenCalledWith('task:incoming', expect.any(Function));
+    });
+  });
+
+  it('should clean up task:incoming event on unmount', async () => {
+    const {unmount} = renderHook(() => useTaskList({cc: ccMock}));
+
+    // Simulate unmount
+    unmount();
+
+    waitFor(() => {
+      expect(ccMock.off).toHaveBeenCalledWith('task:incoming', expect.any(Function));
+    });
+  });
+
+  it('should not register event multiple times', async () => {
+    const {result} = renderHook(() => useTaskList({cc: ccMock}));
+
+    // Trigger TASK_INCOMING event once
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock);
+    });
+
+    // Simulate triggering the event again (should not register again)
+    act(() => {
+      ccMock.on.mock.calls[0][1](taskMock);
+    });
+
+    waitFor(() => {
+      expect(ccMock.on).toHaveBeenCalledTimes(1); // Event should only be registered once
+      expect(ccMock.off).toHaveBeenCalledTimes(1); // Event cleanup should still happen once
+    });
+  });
```

Comment:
Tests look good, we are just not checking any error scenarios here, please see if we can add a few tests for it. **If possible**